File: DataStructure/DataStructure_CPP/2.Stack and Queue/Stack.cc
#include<iostream>
using namespace std;
template<class T>

class Stack{

private:
    struct Node{
    public:
        T item;
        Node *next;
    };
    Node *top = NULL;
    T i;
public:
    void push(T i){
         Node *oldtop = top;
        top = new Node();
        top->item = i;
        top->next = oldtop;

    };
    void pop(){
        Node *p = top;
        top = top->next;
        delete p;
    };    
};

int main(){
    Stack<int> *s = new Stack<int>();

}

File: Algorithms/w2_stack_queue/Stack/Stack.java
package Stack;
public class Stack<T>{
    private class Node<T>{
        T[] item;
        Node<T> next;
    }
    private Node<T> top = null;
    

    private void push(T[] i){
        Node<T> oldtop = top;
        top = new Node<T>();
        top.item = i;
        top.next = oldtop;
        

    }
    private void pop(){
        top = top.next;

    }


}